test/integration/src/test/java/io/pravega/test/integration/BatchClientTest.java

```diff
@@ -173,6 +178,38 @@ public void testBatchClient() throws Exception {
         ArrayList<String> dataAtOffset = Lists.newArrayList(seg0Iterator);
         assertEquals(1, dataAtOffset.size());
         assertEquals(DATA_OF_SIZE_30, dataAtOffset.get(0));
+
```

Comment:
I would suggest to write this new code in a new test method (e.g., testBatchClientSegmentTruncation), given that it specifically verifies a functionality related to a particular bug. In case of future failures, it would be easier to locate the problem.

Reply:
done.